lib/msal-common/src/client/BaseClient.ts

```diff
@@ -126,6 +127,25 @@ export abstract class BaseClient {
                     break;
             }
         }
+
+        // Add client app configuration headers
+        const extraClientConfigTokenRequestHeaders =
+            this.config.authOptions.extraTokenRequestHeaders;
+
+        if (extraClientConfigTokenRequestHeaders) {
+            for (const headerName in extraClientConfigTokenRequestHeaders) {
+                headers[headerName] =
+                    extraClientConfigTokenRequestHeaders[headerName];
```

Comment:
Do we want to add any checking to ensure that the key/value strings are valid?  No added " or : in there for instance.

Reply:
We certainly can. I didn't add validation initially because we generally take config input as opaque. We can throw ourselves if validation fails instead of letting the engine throw.

Reply:
@peterzenz there's options for this, interested in your input:
1. Validate and throw at the moment client app/config is built
2. Validate and throw when the token/ request is built
3. Validate and just omit invalid headers when the token request is built, log a warning

Reply:
Thanks for the options.  I think I'd be OK to skip this here as is and let the call fail downstream.  Thanks.